tests/lib.rs

```diff
@@ -323,6 +323,149 @@ fn it_works_with_sql_recipe() {
 }
 
 #[test]
+fn it_works_with_simple_arithmetic() {
+    let mut g = distributary::Blender::new();
+    let sql = "
+        CREATE TABLE Car (id int, price int, PRIMARY KEY(id));
+        CarPrice: SELECT 2 * price FROM Car WHERE id = ?;
+    ";
+
+    let recipe = g.migrate(|mig| {
+        let mut recipe = distributary::Recipe::from_str(&sql, None).unwrap();
+        recipe.activate(mig, false).unwrap();
+        recipe
+    });
+
+    let car_index = recipe.node_addr_for("Car").unwrap();
+    let count_index = recipe.node_addr_for("CarPrice").unwrap();
+    let mut mutator = g.get_mutator(car_index);
+    let getter = g.get_getter(count_index).unwrap();
+    let id: distributary::DataType = 1.into();
+    let price: distributary::DataType = 123.into();
+    mutator.put(vec![id.clone(), price]).unwrap();
+
+    // Let writes propagate:
+    sleep();
+
+    // Retrieve the result of the count query:
+    let result = getter.lookup(&id, true).unwrap();
+    assert_eq!(result.len(), 1);
+    assert_eq!(result[0][1], 246.into());
```

Comment:
I think I might have gotten something wrong with the output order here. For other queries it looks like the query key is outputted as the [last element in the result](https://github.com/ekmartin/distributary/blob/28a9dadb8743cc9d18e19b7140ac83a8fdee5b39/tests/lib.rs#L320-L322) - whereas with arithmetic expressions it seems to be placed first, which I think might be wrong?

Reply:
Never mind - this seems to match the output order for literals (and I think it makes sense since the columns are just outputted in groups like mentioned in https://github.com/mit-pdos/distributary/pull/30#issuecomment-334186613)